1.2.0 Build Fails

I attempted to update our project's dependencies, along with node-canvas to 1.2.0, but the build failed locally on Windows, and thinking that it may have been because of Windows, I pushed it to travis, and it continued to fail there. I would pastebin the entire Travis log, but it doesn't keep new lines, so this is the best I can do, with the compile error:

http://pastebin.com/wzE5eSK9

Thanks!

Hi guys, I've dot the same error here on a red hat machine:

**npm i canvas**
```
make: Entering directory `/home/build/jenkins/jobs/build-rpm/workspace/node_modules/canvas/build'
  SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node
  SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node: Finished
  COPY Release/canvas-postbuild.node
  CXX(target) Release/obj.target/canvas/src/Canvas.o
In file included from ../src/Canvas.cc:8:
../src/PNG.h: In function ‘cairo_status_t canvas_write_png(cairo_surface_t*, void (*)(png_struct*, png_byte*, png_size_t), void*)’:
../src/PNG.h:165: warning: case label value is less than minimum value for type
  CXX(target) Release/obj.target/canvas/src/CanvasGradient.o
  CXX(target) Release/obj.target/canvas/src/CanvasPattern.o
  CXX(target) Release/obj.target/canvas/src/CanvasRenderingContext2d.o
../src/CanvasRenderingContext2d.cc: In member function ‘void Context2d::shadow(void (*)(cairo_t*))’:
../src/CanvasRenderingContext2d.cc:378: error: ‘cairo_surface_create_similar_image’ was not declared in this scope
make: *** [Release/obj.target/canvas/src/CanvasRenderingContext2d.o] Error 1
make: Leaving directory `/home/build/jenkins/jobs/build-rpm/workspace/node_modules/canvas/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:810:12)
gyp ERR! System Linux 2.6.32-431.29.2.el6.x86_64
gyp ERR! command "node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/build/jenkins/jobs/build-rpm/workspace/node_modules/canvas
gyp ERR! node -v v0.10.33
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok
npm ERR! Linux 2.6.32-431.29.2.el6.x86_64
npm ERR! argv "node" "/usr/bin/npm" "i" "canvas"
npm ERR! node v0.10.33
npm ERR! npm  v2.4.1
npm ERR! code ELIFECYCLE

npm ERR! canvas@1.2.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the canvas@1.2.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the canvas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls canvas
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/build/jenkins/jobs/build-rpm/workspace/npm-debug.log
```

This error doesn't happen with v1.1.6, so I'll pin our dependencies to that in the mean time. However, the warning message about the case label is also present with 1.1.6:

**npm i canvas@1.1.6**
```
-
> canvas@1.1.6 install /home/build/node_modules/canvas
> node-gyp rebuild

make: Entering directory `/home/build/node_modules/canvas/build'
  SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node
  SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node: Finished
  COPY Release/canvas-postbuild.node
  CXX(target) Release/obj.target/canvas/src/Canvas.o
In file included from ../src/Canvas.cc:8:
../src/PNG.h: In function ‘cairo_status_t canvas_write_png(cairo_surface_t*, void (*)(png_struct*, png_byte*, png_size_t), void*)’:
../src/PNG.h:165: warning: case label value is less than minimum value for type
  CXX(target) Release/obj.target/canvas/src/CanvasGradient.o
  CXX(target) Release/obj.target/canvas/src/CanvasPattern.o
  CXX(target) Release/obj.target/canvas/src/CanvasRenderingContext2d.o
  CXX(target) Release/obj.target/canvas/src/color.o
  CXX(target) Release/obj.target/canvas/src/Image.o
  CXX(target) Release/obj.target/canvas/src/ImageData.o
  CXX(target) Release/obj.target/canvas/src/init.o
  CXX(target) Release/obj.target/canvas/src/PixelArray.o
  CXX(target) Release/obj.target/canvas/src/FontFace.o
  SOLINK_MODULE(target) Release/obj.target/canvas.node
  SOLINK_MODULE(target) Release/obj.target/canvas.node: Finished
  COPY Release/canvas.node
make: Leaving directory `/home/build/node_modules/canvas/build'
npm WARN unmet dependency /home/build/node_modules/bower/node_modules/insight/node_modules/request/node_modules/form-data requires async@'~0.9.0' but will load
npm WARN unmet dependency /home/build/node_modules/bower/node_modules/insight/node_modules/async,
npm WARN unmet dependency which is version 0.2.10
canvas@1.1.6 ../../../../node_modules/canvas
└── nan@1.2.0
```

--- a/src/CanvasRenderingContext2d.cc
+++ b/src/CanvasRenderingContext2d.cc
@@ -372,7 +372,7 @@ Context2d::shadow(void (fn)(cairo_t *cr)) {
     cairo_user_to_device_distance(_context, &dx, &dy);
     int pad = state->shadowBlur * 2;
     cairo_surface_t *surface = cairo_get_group_target(_context);
-    cairo_surface_t *shadow_surface = cairo_surface_create_similar_image(surface,
+    cairo_surface_t *shadow_surface = cairo_image_surface_create(
       CAIRO_FORMAT_ARGB32,
       dx + 2 * pad,
       dy + 2 * pad);

